### PR TITLE
Add to phpunit command support for testsuite option (#1787).

### PR DESCRIPTION
**Pull request by greylabel**
Improves #1787 .

Changes proposed:
- Remove hardcoded <directory> arg `.` and add support for `name` and `file`.
- Add support for `testsuite` option.
- Update docs.

### CODE REVIEW

**Comment by greylabel** [comment]
`project.yaml`:

```
phpunit:
    config: ${docroot}/core/phpunit.xml
    path: '${docroot}/core'
```

Not the differences when `.` is not passed as an arg. Run is significatly faster and looks to be executing more tests.
```
vagrant@local:/var/www/site$ blt -vvv tests:phpunit
[warning] The xDebug extension is loaded. This will significantly decrease performance.
[Filesystem\FilesystemStack] mkdir ["/var/www/site/reports/phpunit"]
[Filesystem\FilesystemStack] touch ["/var/www/site/reports/phpunit/results.xml"]
[Testing\PHPUnit] Running PHPUnit  --log-junit /var/www/site/reports/phpunit/results.xml -v --configuration /var/www/site/docroot/core/phpunit.xml
[Testing\PHPUnit] Running /var/www/site/vendor/bin/phpunit --log-junit /var/www/site/reports/phpunit/results.xml -v --configuration /var/www/site/docroot/core/phpunit.xml in /var/www/site/docroot/core
PHPUnit 4.8.36 by Sebastian Bergmann and contributors.

Runtime:	PHP 7.1.9-1+ubuntu16.04.1+deb.sury.org+1 with Xdebug 2.5.0
Configuration:	/var/www/site/docroot/core/phpunit.xml

...........................................................    59 / 20032 (  0%)
...................................^C
```


When `.` is passed as an arg.

```
vagrant@local:/var/www/site$ blt -vvv tests:phpunit
[warning] The xDebug extension is loaded. This will significantly decrease performance.
[Filesystem\FilesystemStack] mkdir ["/var/www/site/reports/phpunit"]
[Filesystem\FilesystemStack] touch ["/var/www/site/reports/phpunit/results.xml"]
[Testing\PHPUnit] Running PHPUnit  --log-junit /var/www/site/reports/phpunit/results.xml . -v --configuration /var/www/site/docroot/core/phpunit.xml
[Testing\PHPUnit] Running /var/www/site/vendor/bin/phpunit --log-junit /var/www/site/reports/phpunit/results.xml . -v --configuration /var/www/site/docroot/core/phpunit.xml in /var/www/site/docroot/core
PHPUnit 4.8.36 by Sebastian Bergmann and contributors.

Runtime:	PHP 7.1.9-1+ubuntu16.04.1+deb.sury.org+1 with Xdebug 2.5.0
Configuration:	/var/www/site/docroot/core/phpunit.xml

...............FFFF..FFFFEF..F.FF..........................    59 / 19875 (  0%)
.......SSSS......................F^C
vagrant@local:/var/www/site$
```

**Comment by greylabel** [comment]
@grasmash Let me know what you think about this approach.

File: config/build.yml [hunk 1]
```diff
@@ -113,6 +113,8 @@ phpcbf:
 # @todo Move to subkey of tests.
 phpunit:
   - path: '${repo.root}/tests/phpunit'
+    class: 'ExampleTest'
```
Comment (grasmash) [inline config/build.yml]: By default, BLT does not place this file in the project repository. You must first run `blt examples:init` in order to generate the file.

So, I would expect this change to cause `blt tests:phpunit` to fail out of the box.

**Comment by grasmash** [comment]
Ah, I forgot I merged #1910.

**Comment by greylabel** [comment]
@grasmash Would you prefer these PR go against `8.9.x` rather than the `8.x` branch?

**Comment by dpagini** [comment]
@grasmash & @greylabel not sure if you'd see this, but... if this got merged against 8.x, what release would I expect to see this functionality in? I don't think it's in `8.9.10` and I also don't see these changes in the `9.x` branch either.

**Comment by greylabel** [comment]
@dpagini it's now in `9.1.x`.

**Comment by danepowell** [comment]
Yep, 8.x basically became 9.1.x. All enhancements are going into 9.1.x.

**Comment by dpagini** [comment]
Ahh, thank you both!

**Pull request merged**